docs/synapse/devopsguide.rst

```diff
@@ -499,6 +499,110 @@ Once they are enrolled, the user can connect using the Telepath URL ``aha://cort
+
+``complexity``
+    The complexity key must be set to an object. It can have the following keys:
+
+    ``length``
+        Minimum password character length. This is an integer value.
+
+    ``sequences``
+        Maximum sequence length in a password. Sequences can be letters or number, forward or reverse. This is an
+        integer value.
+
+    ``upper:count``
+        The minimum number of uppercase characters required in password. This is a string value.
+
+    ``upper:valid``
+        All valid uppercase characters. This defaults to a string of uppercase ASCII characters. This can be set to
+        a null value to disable any checking of the uppercase characters rules.
+
+    ``lower:count``
+        The minimum number of lowercase characters required in password.
+
+    ``lower:valid``
+        All valid lowercase characters. This defaults to a string of lowercase ASCII characters. This can be set to
+        a null value to disable any checking of the lowercase character rules.
+
+    ``special:count``
+        The minimum number of special characters required in password.
+
+    ``special:valid``
+        All valid special characters. This defaults to a string of ASCII punctuation characters. This can be set to
+        a null value to disable any checking of the special characters rules.
+
+    ``number:count``
+        The minimum number of digit characters required in password.
+
+    ``number:valid``
+        All valid digit characters. This defaults to a string of ASCII number characters. This can be set to
+        a null value to disable any checking of the number character rules.
+
+The following example shows setting a password policy on the Cortex with the following policy:
+
+* Maximum of three failed password login attempts before locking.
+
+* Keep the previous two passwords to prevent password reuse.
+
+* Complexity rules:
+
+    * Require at least 12 total characters.
+
+    * Disallow sequences of more than 3 characters in a row.
+
+    * Require at least two uppercase characters.
+
+    * Require at least two lowercase characters.
+
+    * Specify a custom set of lowercase characters to check against (ASCII & some unicode characters).
+
+    * Require at least two special characters.
+
+    * Require at least two numbers.
+
+The following Compose file shows using the policy:
+
+::
+
+    services:
+      00.cortex:
+        user: "999"
+        image: vertexproject/synapse-cortex:v2.x.x
+        network_mode: host
+        restart: unless-stopped
+        volumes:
+            - ./storage:/vertex/storage
+        environment:
+            SYN_CORTEX_AXON: aha://axon...
+            SYN_CORTEX_JSONSTOR: aha://jsonstor...
+            SYN_CORTEX_AUTH_PASSWD_POLICY: '{"complexity": {"length": 12, "sequences": 3, "upper:count": 2, "lower:count": 2, "lower:valid": "abcdefghijklmnopqrstuvwxyz\\u03b1\\u03b2\\u03b3", "special:count": 2, "number:count": 2}, "attempts": 3, "previous": 2}'
```

Comment:
Do the unicode characters have to be \u encoded or can the just be pasted directly?

Reply:
We can embed them directly. That was a default JSON serialization.